src/artifacts.js

```diff
@@ -176,7 +220,31 @@ api.declare({
   var isPublic = /^public\//.test(name);
   var details  = {};
   switch (storageType) {
+    case 'blob':
+      // Generate the details that we'll be using.
+      details.size = input.size;
+      details.sha256 = input.sha256;
+      if (input.parts) {
+        details.parts = input.parts;
+      }
+      // TODO: These ought to be configurable
+      details.provider = 's3'; // MAKECONFIG
+      details.region = 'us-east-1'; // MAKECONFIG
+      if (input.encoding) {
+        details.encoding = input.encoding;
+      }
+      if (isPublic) {
+        details.bucket = 'taskcluster-blobs'; // MAKECONFIG
+      } else {
+        details.bucket = 'taskcluster-private-blobs'; // MAKECONFIG
+      }
+
```

Comment:
Yeah, going back and forth over this... it seems that no matter what we do we'll have a race where we risk creating an `uploadId` too much. Right?

Ie. we could have two racing later down in the `.modify` section... in fact it's probaby problematic to just blindly overwrite it... concurrency in the morning is hard. So only 92% sure at this point :)

Regardless, we're not going to race very often, so really however we resolve it doesn't matter... because it'll only happen every odd moon (is that an expression?).